flytepropeller/pkg/controller/nodes/cache.go

```diff
@@ -216,7 +216,10 @@ func (n *nodeExecutor) WriteCatalogCache(ctx context.Context, nCtx interfaces.No
 		catalogKey.Identifier.Domain, catalogKey.Identifier.Name, catalogKey.Identifier.Version)
 
 	outputPaths := ioutils.NewReadOnlyOutputFilePaths(ctx, nCtx.DataStore(), nCtx.NodeStatus().GetOutputDir())
-	outputReader := ioutils.NewRemoteFileOutputReader(ctx, nCtx.DataStore(), outputPaths, 0)
+	outputReader, err := ioutils.NewRemoteFileOutputReader(ctx, nCtx.DataStore(), outputPaths, 0)
+	if err != nil {
+		return catalog.NewStatus(core.CatalogCacheStatus_CACHE_DISABLED, nil), errors.Wrapf(err, "failed to initialize the remote output file reader")
```

Comment:
Is my understanding correct that when passing strategy `0`, we can't get an error?

Reply:
That is true.

Reply:
Do you think it would be possible to not pass the strategy to `NewRemoteFileOutputReader` given that below you use a new function `ioutils.NewRemoteFileOutputReaderWithErrorAggregationStrategy`?

Avoiding this would allow us to not modify this file and several others at all where currently no errors can occur when constructing the output reader.

Reply:
We can do that if you feel uncomfortable with these changes, but I felt like the no error nature of the `NewRemoteFileOutputReader` was more of an oversight.